### PR TITLE
Nuklear function override

### PR DESCRIPTION
**Pull request by Nuthen**

### CODE REVIEW

File: states/splash.lua [hunk 1]
```diff
@@ -120,4 +120,11 @@ function splash:draw()
     print_with_shadow(Shaderview._VERSION, version_x, version_y)
 end
 
+-- set stub functions for any unused handlers
+for handler, fn in pairs(love.handlers) do
+    if not splash[handler] then
+        splash[handler] = function(...) end
+    end
+end
+
```
Comment (camchenry) [inline states/splash.lua]: I would prefer to see this go away and instead check that the callbacks exists in the first place before calling them.

File: main.lua [hunk 2]
```diff
@@ -107,8 +107,11 @@ OS: %s
 
     -- Draw is left out so we can override it ourselves
     local callbacks = {'errhand', 'update'}
+    local leftoutCallbacks = {'keypressed', 'keyreleased', 'mousepressed', 'mousereleased', 'mousemoved', 'textinput', 'wheelmoved'}
```
Comment (camchenry) [inline main.lua]: Really minor nitpick: I'm using underscores in this project (sort of an experiment) so it should be something like left_out_callbacks

File: main.lua [hunk 3]
```diff
@@ -143,34 +146,63 @@ function love.draw()
...
-    Nuklear.keypressed(key, code, isRepeat)
     if not RELEASE and code == CONFIG.debug.key then
         DEBUG = not DEBUG
     end
+
+    if Nuklear.keypressed(key, code, isRepeat) then
+        return
+    end
+
+    State.current():keypressed(key, code, isRepeat)
```
Comment (camchenry) [inline main.lua]: I think it will be more maintainable in the future to just check if the current state has the callback in the first place.

**Review by camchenry — CHANGES_REQUESTED** [review]

**Comment by Nuthen** [comment]
I have made relevant changes.

**Comment by camchenry** [comment]
I'm not using Nuklear anymore, as you know, so I'm going to close this because it's no longer relevant.

**Pull request closed without merging**